ikarus/finiteelements/fetraits.hh

```diff
@@ -32,11 +33,17 @@ struct FETraits
   /** \brief Type of the flat basis */
   using FlatBasis = typename Basis::FlatBasis;
 
+  /** \brief Type of the untouched basis */
+  using UntouchedBasis = typename Basis::UntouchedBasis;
+
+  /** \brief Type of the basis version*/
+  using BasisVersion = std::conditional_t<useFlat, FlatBasis, UntouchedBasis>;
```

Comment:
Maybe stick to `Basis` as name, since most of the code should care about what type and this is the best one

Reply:
`Basis` already existed there which is the type of the global basis passed to a finite element. Hence I used BasisVersion here. Renaming it to `Basis` might then create confusions in finite elements. What about just `PreBasis`?

Reply:
It is no PreBasis, maybe we have to rename the Ikarus::Basis to Ikarus::BasisHandler to not confuse it with the Dune::Basis stuff?

Reply:
FETraits will get BH for basis handler?

Reply:
I am not renaming the `makeBasis` function to `makeBasisHandler` here, may be it can be done separately. I have renamed the file to `basishandler.hh`